src/app/Scenes/Artwork/Components/CommercialButtons/BuyNowButton.tsx

```diff
@@ -127,7 +129,11 @@ export const BuyNowButton = ({
       width={100}
       haptic
     >
-      {renderSaleMessage && saleMessage ? `Purchase ${saleMessage}` : "Purchase"}
+      {!!buttonText
+        ? buttonText
+        : renderSaleMessage && saleMessage
+          ? `Purchase ${saleMessage}`
+          : "Purchase"}
```

Comment:
I would try to avoid double `? :` whenever possible, as they are quite difficult to read, especially in combination with an additional `&&`.

```suggestion
const displaySaleMessage = ....

.....
      {!!buttonText
        ? buttonText
        : displaySaleMessage}
```